docs/spec/openapi.yaml

```diff
@@ -285,8 +285,20 @@ components:
         completed_at:
           type: string
           format: date-time
+        role:
+          description: |
+            Specifies the sender of the message. Allowed values are:
+            - `"user"` for messages sent by an end-user.
+            - Any other string indicating the specific name of the agent sending the message.
+          oneOf:
+            - type: string
+              enum: [user]
+            - type: string
+              description: Name of the agent.
+              example: summarizer_agent
```

Comment:
Special-casing of the string `"user"` seems ugly since then we need to prohibit an agent named `"user"` etc. -- IMO a better solution would be prefixing the agent names, so it's either `"user"` or `"agent/{agent_name}"` which avoids the ambiguity.